solr/core/src/java/org/apache/solr/cloud/overseer/ZkStateWriter.java

```diff
@@ -210,6 +265,14 @@ public ClusterState writePendingUpdates() throws IllegalStateException, KeeperEx
           } else if (c.getStateFormat() == 1) {
             isClusterStateModified = true;
           }
+          if(cmd.ops != null && !cmd.ops.isPreOp()) {
```

Comment:
Does this need a `&& !cmd.ops.get().isEmpty()`  to match the pre ops?

Reply:
well the `isEmpty()` check can be removed in the pre-ops also as the ` PerReplicaStates.presist()` checks them anyway